test/test.py

```diff
@@ -146,45 +145,42 @@ def run_single_test(self, dirname=None, config_path=None, force=False):
                     config_path,
                     '\n'.join(diff_lines)))
 
-class ExecContext:
-    """Creates temporary config, data file and lets you call r2e with them
-    easily. Cleans up temp files afterwards.
-
-    Example:
-
-    with ExecContext(config="[DEFAULT]\nto=me@example.com") as context:
-        context.call("run", "--no-send")
-
-    """
-    def __init__(self, config):
-        self.tmpdir = tempfile.mkdtemp()
-        self.cfg_path = _os.path.join(self.tmpdir, "rss2email.cfg")
-        self.data_path = _os.path.join(self.tmpdir, "rss2email.json")
-
-        with open(self.cfg_path, "w") as f:
-            f.write(config)
-
-    def __enter__(self):
-        return self
-
-    def __exit__(self, type, value, traceback):
-        for path in [self.cfg_path, self.data_path]:
-            if _os.path.exists(path):
-                _os.remove(path)
-        _os.rmdir(self.tmpdir)
-
-    def call(self, *args):
-        subprocess.call([r2e_path, "-c", self.cfg_path, "-d", self.data_path] + list(args))
-
 class NoLogHandler(http.server.SimpleHTTPRequestHandler):
     "No logging handler serving test feed data from test_dir"
-    def translate_path(self, path):
-        path = http.server.SimpleHTTPRequestHandler.translate_path(self, path)
-        return _os.path.join(test_dir, path)
+
+    if sys.version_info >= (3, 7):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs, directory=test_dir)
+    else:
+        def translate_path(self, path):
+            cwd = _os.getcwd()
+            try:
+                _os.chdir(test_dir)
+                return super().translate_path(path)
+            finally:
+                _os.chdir(cwd)
 
     def log_message(self, format, *args):
         return
 
+def webserver_for_test_fetch(queue, num_requests, wait_time):
```

Comment:
Is there some reason you moved this out of the test? Since you had to add `for_test_fetch` to the function name, this seems to indicate to me that it really belongs inside `TestFetch`.

Reply:
Yes, there is a reason. I encountered an error on running the test. I wrote it in the [commit](https://github.com/rss2email/rss2email/pull/112/commits/5b588dfc4b6df06ebdea48baa8e9367ac68a6b57) message:
~~~
AttributeError: Can't pickle local object 'TestFetch.test_delay.<locals>.webserver'
~~~

Maybe this error is specific for Windows or for Python 3.8, I have not investigated this. But moving the `webserver` function out from the locals helped.

Reply:
This is interesting, macOS Python 3.8 gets the same error. I've seen at least one other similar issue: https://github.com/pytorch/pytorch/issues/36515. Something to do with multiprocessing changing in 3.8 from using fork to spawn on macOS and I guess this probably never worked on Windows.